modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java

```diff
@@ -0,0 +1,852 @@
+        try {
+            PartitionAllocationMap allocationMap = ctx.partitionStatMap();
+
+            allocationMap.prepareForSnapshot();
+
+            for (GroupPartitionId pair : parts) {
+                PagesAllocationRange allocRange = allocationMap.get(pair);
+
+                GridDhtLocalPartition part = pair.getPartitionId() == INDEX_PARTITION ? null :
+                    cctx.cache()
+                        .cacheGroup(pair.getGroupId())
+                        .topology()
+                        .localPartition(pair.getPartitionId());
+
+                // Partition can be reserved.
+                // Partition can be MOVING\RENTING states.
+                // Index partition will be excluded if not all partition OWNING.
+                // There is no data assigned to partition, thus it haven't been created yet.
+                assert allocRange != null || part == null || part.state() != GridDhtPartitionState.OWNING :
+                    "Partition counters has not been collected " +
+                        "[pair=" + pair + ", snpName=" + snpName + ", part=" + part + ']';
+
+                if (allocRange == null) {
+                    List<GroupPartitionId> missed = parts.stream()
+                        .filter(allocationMap::containsKey)
+                        .collect(Collectors.toList());
+
+                    acceptException(new IgniteCheckedException("Snapshot operation cancelled due to " +
+                        "not all of requested partitions has OWNING state [missed=" + missed + ']'));
+
+                    break;
+                }
+
+                PageStore store = ((FilePageStoreManager)cctx.pageStore()).getStore(pair.getGroupId(), pair.getPartitionId());
+
+                partFileLengths.put(pair, store.size());
+                partDeltaWriters.get(pair).init(allocRange.getCurrAllocatedPageCnt());
+            }
+        }
+        catch (IgniteCheckedException e) {
+            acceptException(e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onCheckpointBegin(Context ctx) {
+        if (stopping.getAsBoolean())
+            return;
+
+        // Snapshot task is now started since checkpoint writelock released.
+        if (!startedFut.onDone())
+            return;
+
+        // Submit all tasks for partitions and deltas processing.
+        List<CompletableFuture<Void>> futs = new ArrayList<>();
+        FilePageStoreManager storeMgr = (FilePageStoreManager)cctx.pageStore();
+
+        if (log.isInfoEnabled())
+            log.info("Submit partition processings tasks with partition allocated lengths: " + partFileLengths);
+
+        // Process binary meta.
+        futs.add(CompletableFuture.runAsync(
+            wrapExceptionIfStarted(() ->
+                    snpSndr.sendBinaryMeta(cctx.kernalContext()
+                        .cacheObjects()
+                        .metadata(Collections.emptyList()))),
```

Comment:
Use `metadata()` method here, it's more lightweight.

Reply:
Fixed.